[mssql-server-windows] Attaching a db from files stored in Azure page blobs

perrysk-msft opened the pull request:
Based on a couple of requests to be able to attach a database from files stored in Azure Blobs, we made the following code changes to extend the **attach_dbs** environment variable and support this scenario. At a high level, we added saskey to be able to create a sql credential before attaching the db from Azure page blobs (based on this: https://docs.microsoft.com/en-us/sql/relational-databases/databases/sql-server-data-files-in-microsoft-azure article). 

Let us know what you think - We are interested in your feedback before merging the change and building a new Docker image.

twright-msft commented:
A few suggestions:

- Rename $container to $blob_container to disambiguate what is going on here a little bit.  When I first saw this I thought 'container' as in Docker container. :)

- The saskey property existence check is done at the DB level but then the key is added for each file.  Is that intentional?  Should we do this once/DB?

- Something seems off about the indentation.  Can you please double check that?

perrysk-msft commented:
@twright-msft - Thanks for the suggestions! I updated the PR to address the feedback.

- I renamed $container to $blob_container
- The saskey property existence check at the DB level is intentional as the data and log files might be stored in different blob containers. 
- I believe that the indentation is now fixed.

Thanks, Perry
